app/controllers/employments/UpdateEmploymentController.scala

```diff
@@ -160,86 +160,90 @@ class UpdateEmploymentController @Inject() (
       )
       .bindFromRequest()
       .fold(
-        formWithErrors =>
-          journeyCacheService.currentCache map { currentCache =>
-            implicit val user: AuthedUser = request.taiUser
-            BadRequest(
-              canWeContactByPhone(
-                Some(user),
-                telephoneNumberViewModel(currentCache(UpdateEmploymentConstants.EmploymentIdKey).toInt),
-                formWithErrors
-              )
-            )
-          },
-        form => {
-          val mandatoryData = Map(
-            UpdateEmploymentConstants.TelephoneQuestionKey -> Messages(
-              s"tai.label.${form.yesNoChoice.getOrElse(FormValuesConstants.NoValue).toLowerCase}"
-            )
+        formWithErrors => {
+          val employmentId = request.userAnswers.get(UpdateEmploymentIdPage)
+          implicit val user: AuthedUser = request.taiUser
+          Future.successful(
+            BadRequest(canWeContactByPhone(Some(user), telephoneNumberViewModel(employmentId.get), formWithErrors))
           )
-          val dataForCache = form.yesNoChoice match {
+        },
+        form => {
+          val yesNoTask = form.yesNoChoice match {
             case Some(yn) if yn == FormValuesConstants.YesValue =>
-              mandatoryData ++ Map(UpdateEmploymentConstants.TelephoneNumberKey -> form.yesNoTextEntry.getOrElse(""))
-            case _ => mandatoryData ++ Map(UpdateEmploymentConstants.TelephoneNumberKey -> "")
+              journeyCacheNewRepository.set(
+                request.userAnswers
+                  .setOrException(
+                    UpdateEmploymentTelephoneQuestionPage,
+                    Messages(s"tai.label.${form.yesNoChoice.getOrElse(FormValuesConstants.NoValue).toLowerCase}")
```

Comment:
Shouldn't this be YesValue?
Also lines 173-180 are almost identical to lines 182 - 190 so could make more concise by moving out of case match and parameterising